src/trunk.c

```diff
@@ -5445,8 +5450,10 @@ trunk_compact_bundle(void *arg, void *scratch_buf)
          key_string(trunk_data_config(spl), end_key),
          req->height,
          req->bundle_no);
+      key_buffer_copy_key(&req->start_key, key_buffer_key(&scratch->req_original_start_key));
```

Comment:
Restore it here so that the filter compaction will cover the entire range of the original bundle compaction request,